client/my-sites/upgrades/domain-search/site-redirect.jsx

```diff
@@ -1,63 +1,86 @@
+import { connect } from 'react-redux';
+import { localize } from 'i18n-calypso';
 
 /**
  * Internal dependencies
  */
-var HeaderCake = require( 'components/header-cake' ),
-	Main = require( 'components/main' ),
-	SiteRedirectStep = require( './site-redirect-step' ),
-	observe = require( 'lib/mixins/data-observe' );
+import HeaderCake from 'components/header-cake';
+import Main from 'components/main';
+import SiteRedirectStep from './site-redirect-step';
+import isSiteUpgradeable from 'state/selectors/is-site-upgradeable';
+import { getSelectedSite, getSelectedSiteId, getSelectedSiteSlug } from 'state/ui/selectors';
+import QueryProductsList from 'components/data/query-products-list';
 
-var SiteRedirect = React.createClass( {
-	mixins: [ observe( 'productsList', 'sites' ) ],
+class SiteRedirect extends Component {
+	static propTypes = {
+		cart: PropTypes.object.isRequired,
+		selectedSite: PropTypes.object.isRequired,
+		selectedSiteSlug: PropTypes.string.isRequired,
+		isSiteUpgradeable: PropTypes.func.isRequired,
+		productsList: PropTypes.object.isRequired,
+		translate: PropTypes.func.isRequired,
+	};
 
-	propTypes: {
-		productsList: React.PropTypes.object.isRequired,
-		sites: React.PropTypes.object.isRequired
-	},
+	handleBackToDomainSearch = () => {
+		page( '/domains/add/' + this.props.selectedSiteSlug );
+	};
 
-	componentWillMount: function() {
-		this.checkSiteIsUpgradeable();
-	},
-
-	componentDidMount: function() {
-		this.props.sites.on( 'change', this.checkSiteIsUpgradeable );
-	},
-
-	componentWillUnmount: function() {
-		this.props.sites.off( 'change', this.checkSiteIsUpgradeable );
-	},
+	componentDidMount() {
+		this.checkSiteIsUpgradeable( this.props );
+	}
 
-	checkSiteIsUpgradeable: function( ) {
-		var selectedSite = this.props.sites.getSelectedSite();
+	componentWillReceiveProps( nextProps ) {
+		if ( nextProps.selectedSiteId !== this.props.selectedSiteId ) {
+			this.checkSiteIsUpgradeable( nextProps );
+		}
+	}
 
-		if ( selectedSite && ! selectedSite.isUpgradeable() ) {
+	checkSiteIsUpgradeable( props ) {
+		if ( ! props.isSiteUpgradeable ) {
 			page.redirect( '/domains/add' );
 		}
-	},
+	}
+
+	handleBackToDomainSearch() {
+		page( '/domains/add/' + this.props.selectedSiteSlug );
+	}
 
-	backToDomainSearch: function() {
-		page( '/domains/add/' + this.props.sites.getSelectedSite().slug );
-	},
+	render() {
+		const {
+			cart,
+			selectedSite,
+			productsList,
+			translate,
+		} = this.props;
 
-	render: function() {
 		return (
 			<Main>
-				<HeaderCake onClick={ this.backToDomainSearch }>
-					{ this.translate( 'Redirect a Site' ) }
+				<QueryProductsList />
+
+				<HeaderCake onClick={ this.handleBackToDomainSearch }>
+					{ translate( 'Redirect a Site' ) }
 				</HeaderCake>
 
 				<SiteRedirectStep
-					cart={ this.props.cart }
-					products={ this.props.productsList.get() }
-					selectedSite={ this.props.sites.getSelectedSite() } />
+					cart={ cart }
+					products={ productsList }
```

Comment:
It's for another iteration, but `products` and  `selectedSite` can be connected in  `<SiteRedirectStep />`. Redux shines here!